plugin/evm/vm_test.go

```diff
@@ -206,8 +207,7 @@ func setupGenesis(
 	ctx.Keystore = userKeystore.NewBlockchainKeyStore(ctx.ChainID)
 
 	issuer := make(chan commonEng.Message, 1)
-	prefixedDB := prefixdb.New([]byte{1}, baseDB)
-	return ctx, prefixedDB, genesisBytes, issuer, atomicMemory
+	return ctx, baseDB, genesisBytes, issuer, atomicMemory
```

Comment:
IMO this is not entirely correct since it's returning the baseDB which has a prefixDB defined on top of it, let's either return a new memdb or retain the prefixDB so the atomicMemory is separate. 
This could be an issue in iteration if the atomic memory is used

Reply:
In the regression test we should check the `baseDB` rather than `prefixedDB` where the `baseDB` is used in `sharedMemory` and we test out if `sharedMemory` writes to the `baseDB`.